projects/nx-verdaccio/src/plugin/targets/environment.targets.unit-test.ts

```diff
@@ -0,0 +1,463 @@
+import {
+  afterEach,
+  beforeEach,
+  describe,
+  expect,
+  type Mock,
+  type MockInstance,
+} from 'vitest';
+import {
+  type ProjectConfiguration,
+  type TargetConfiguration,
+} from '@nx/devkit';
+
+import {
+  isEnvProject,
+  verdaccioTargets,
+  getEnvTargets,
+  updateEnvTargetNames,
+  VERDACCIO_STORAGE_DIR,
+  TARGET_ENVIRONMENT_VERDACCIO_START,
+  TARGET_ENVIRONMENT_VERDACCIO_STOP,
+  TARGET_ENVIRONMENT_BOOTSTRAP,
+  TARGET_ENVIRONMENT_INSTALL,
+  TARGET_ENVIRONMENT_PUBLISH_ONLY,
+  TARGET_ENVIRONMENT_SETUP,
+  TARGET_ENVIRONMENT_TEARDOWN,
+  TARGET_ENVIRONMENT_E2E,
+} from './environment.targets';
+import {
+  TARGET_PACKAGE_INSTALL,
+  TARGET_PACKAGE_PUBLISH,
+} from './package.targets';
+
+import { PACKAGE_NAME } from '../constants';
+import { type NormalizedCreateNodeOptions } from '../normalize-create-nodes-options';
+
+import { EXECUTOR_ENVIRONMENT_KILL_PROCESS } from '../../executors/kill-process/constant';
+import {
+  EXECUTOR_ENVIRONMENT_BOOTSTRAP,
+  VERDACCIO_REGISTRY_JSON,
+} from '../../executors/env-bootstrap/constants';
+import { EXECUTOR_ENVIRONMENT_TEARDOWN } from '../../executors/env-teardown/constants';
+import { EXECUTOR_ENVIRONMENT_SETUP } from '../../executors/env-setup/constants';
+
+import * as nodePathModule from 'node:path';
+import * as uniquePortModule from '../../executors/env-bootstrap/unique-port';
+
+const JOIN_RESULT = 'mocked-join';
+const PROJECT_NAME = 'unit-test-project';
+const ENVIRONMENTS_DIRECTORY = '/environments';
+const TARGET_NAMES = ['e2e'];
+const TAGS = ['env:production'];
+const TARGETS: { [targetName: string]: TargetConfiguration } = {
+  e2e: { executor: 'nx:test', options: {} },
+  build: { executor: 'nx:build', options: {} },
+};
+const PROJECT_CONFIG: ProjectConfiguration = {
+  root: 'mock-root',
+  name: PROJECT_NAME,
+  targets: TARGETS,
+  tags: [...TAGS],
+};
+const OPTIONS: NormalizedCreateNodeOptions['environments'] = {
+  environmentsDir: ENVIRONMENTS_DIRECTORY,
+  targetNames: TARGET_NAMES,
+  filterByTags: TAGS,
+};
+
+describe('isEnvProject', (): void => {
+  it('should return false if targets are missing', (): void => {
+    const config = { ...PROJECT_CONFIG, targets: null };
+    const result = isEnvProject(config, OPTIONS);
+    expect(result).toBe(false);
+  });
+
+  it('should return false if targetNames are missing', (): void => {
+    const options = { ...OPTIONS, targetNames: null };
+    const result = isEnvProject(PROJECT_CONFIG, options);
+    expect(result).toBe(false);
```

Comment:
The const's could be inlined
```suggestion
    expect(isEnvProject(PROJECT_CONFIG, { ...OPTIONS, targetNames: null })).toBe(false);
```